src/sql/parts/connection/common/connectionManagementService.ts

```diff
@@ -1346,6 +1351,46 @@ export class ConnectionManagementService implements IConnectionManagementService
 		return matchingGroup.color;
 	}
 
+	/**
+	 * Get the current global connection, which is the connection from the active editor, unless OE
+	 * is focused or there is no such editor, in which case it comes from the OE selection. Returns
+	 * undefined when there is no such connection.
+	*/
+	public getCurrentGlobalConnection(): IConnectionProfile {
+		let connection: IConnectionProfile;
+
+		let objectExplorerSelection = this._objectExplorerService.getSelectedProfileAndDatabase();
+		if (objectExplorerSelection) {
+			let objectExplorerProfile = objectExplorerSelection.profile;
+			if (this.isProfileConnected(objectExplorerProfile)) {
+				if (objectExplorerSelection.databaseName) {
+					connection = objectExplorerProfile.cloneWithDatabase(objectExplorerSelection.databaseName);
+				} else {
+					connection = objectExplorerProfile;
+				}
+			}
+			if (this._objectExplorerService.isFocused()) {
+				return connection;
+			}
+		}
+
+		let activeInput = this._editorService.getActiveEditorInput();
+		if (activeInput) {
+			if (activeInput instanceof QueryInput || activeInput instanceof EditDataInput || activeInput instanceof DashboardInput) {
+				connection = this.getConnectionProfile(activeInput.uri);
+			}
+		}
+
+		return connection;
+	}
+
+	public registerObjectExplorerService(service: IObjectExplorerService): void {
```

Comment:
not sure if I understand the reason for this. can you please add comment for the method